### ISSUE TITLE
ZSTD EarlyAbort

### ISSUE THREAD

**Issue by QORTEC** [koverstreet/bcachefs#482]
Recently [openZFS](openzfs/zfs#13244) has added ZSTD EarlyAbort. 

I haven't seen bcachefs mention EarlyAbort anywhere, so I thought I would bring it up.

EarlyAbort allows the compression algorithm to attempt to compress data, and when it deems that the file is not able to be compressed, it will then bypass the compression and just store the file normally.

This has the benefits of not wasting system resources on trying to compress non compressible data.

**Comment by chayleaf** [comment]
Currently, zstd level 3 is used in bcachefs, which is fast enough for this to not be necessary.

**Comment by QORTEC** [comment]
Agreed, though feature would enable using a higher level of compression while minimizing the disadvantages.

**Comment by koverstreet** [comment]
Someone check if the kernel ZSTD implementation supports this, and if it does I'll add it. We recently got compression levels.

**Comment by Komzpa** [comment]
I checked the current kernel-side zstd surface used here. `include/linux/zstd.h` is still a minimal wrapper around the upstream API, and the exported compression parameter surface used by `linux/zstd_compress_module.c` is the normal window/hash/chain/search/minMatch/targetLength/strategy plus frame flags.

I do not see an EarlyAbort-style symbol or parameter exposed to bcachefs today, and `fs/data/compress.c` is using the regular `zstd_get_params()` / `zstd_compress_cctx()` path. So this still looks blocked on exposing/supporting that knob in the kernel zstd API before bcachefs can wire it up.
